### PR TITLE
Add safetensors SD 1.5 tensor names mapping

### PR DESCRIPTION
**Pull request by yichunk**
- Add tensor names mapping from original SD 1.5 huggingface repo https://huggingface.co/runwayml/stable-diffusion-v1-5
- Fix some miscellaneous typos

BUG=b/350994822

### CODE REVIEW

File: ai_edge_torch/generative/examples/stable_diffusion/README.md [hunk 1]
```diff
@@ -2,13 +2,48 @@
...
+python ai_edge_torch/generative/examples/stable_diffusion/convert_to_tflite.py \
+--clip_ckpt=$HOME/stable-diffusion-v1-5/v1-5-pruned-emaonly.safetensors \
+--diffusion_ckpt=$HOME/stable-diffusion-v1-5/v1-5-pruned-emaonly.safetensors \
+--decoder_ckpt=$HOME/stable-diffusion-v1-5/v1-5-pruned-emaonly.safetensors \
+--output_dir=/tmp/stable_diffusion_safetensors/ \
+--ckpt_format=safetensors
+```
+
+
+### PyTorch model checkpoints from third party
```
Comment (haozha111) [inline ai_edge_torch/generative/examples/stable_diffusion/README.md:20]: let's just remove this and only support safetensors format.

File: ai_edge_torch/generative/examples/stable_diffusion/README.md [hunk 2]
```diff
@@ -2,13 +2,48 @@
 This example shows how to use the Edge Generative API to convert a PyTorch Stable Diffusion v1.5 model to TFLite model, and run the image generation inference.
 
 ## Convert PyTorch to TFLite model
+The example provides two source checkpoints mapping. One is original HuggingFace repo, and the other one is a third party PyTorch implementation of Stable Diffusion. Users need to specify the source checkpoints mapping format for conversion script.
```
Comment (haozha111) [inline ai_edge_torch/generative/examples/stable_diffusion/README.md:5]: i guess it's better to only support safetensors for now..

File: ai_edge_torch/generative/examples/stable_diffusion/README.md (hunk 1 above)
Comment (yichunk) [reply ai_edge_torch/generative/examples/stable_diffusion/README.md]: Done

File: ai_edge_torch/generative/examples/stable_diffusion/README.md (hunk 2 above)
Comment (yichunk) [reply ai_edge_torch/generative/examples/stable_diffusion/README.md]: Done

File: ai_edge_torch/generative/examples/stable_diffusion/clip.py [hunk 3]
```diff
@@ -22,10 +22,9 @@
 import ai_edge_torch.generative.layers.model_config as cfg
 import ai_edge_torch.generative.utilities.loader as loading_utils
 
-TENSOR_NAMES = loading_utils.ModelLoader.TensorNames(
+PYTORCH_TENSOR_NAMES = loading_utils.ModelLoader.TensorNames(
```
Comment (haozha111) [inline ai_edge_torch/generative/examples/stable_diffusion/clip.py:25]: do we still need this?

File: ai_edge_torch/generative/examples/stable_diffusion/decoder.py [hunk 4]
```diff
@@ -22,7 +22,7 @@
 import ai_edge_torch.generative.layers.unet.model_config as unet_cfg
 import ai_edge_torch.generative.utilities.stable_diffusion_loader as stable_diffusion_loader
 
-TENSORS_NAMES = stable_diffusion_loader.AutoEncoderModelLoader.TensorNames(
+PYTORCH_TENSOR_NAMES = stable_diffusion_loader.AutoEncoderModelLoader.TensorNames(
```
Comment (haozha111) [inline ai_edge_torch/generative/examples/stable_diffusion/decoder.py:25]: same here..

File: ai_edge_torch/generative/examples/stable_diffusion/convert_to_tflite.py [hunk 5]
```diff
@@ -24,38 +26,92 @@
...
+    required=True,
+)
+arg_parser.add_argument(
+    '--ckpt_format',
+    type=str,
+    choices=['pytorch', 'safetensors'],
+    default='safetensors',
+    nargs='?',
+    const='safetensors',
+)
```
Comment (haozha111) [inline ai_edge_torch/generative/examples/stable_diffusion/convert_to_tflite.py:60]: if we just support one, we can remove this argument

File: ai_edge_torch/generative/examples/stable_diffusion/convert_to_tflite.py [hunk 6]
```diff
@@ -24,38 +26,92 @@
...
+arg_parser.add_argument(
+    '--output_dir',
+    type=str,
+    help='Path to the converted TFLite directory.',
```
Comment (haozha111) [inline ai_edge_torch/generative/examples/stable_diffusion/convert_to_tflite.py:64]: TF Lite

File: ai_edge_torch/generative/examples/stable_diffusion/convert_to_tflite.py [hunk 7]
```diff
@@ -24,38 +26,92 @@
...
+    required=True,
+)
+
 
 @torch.inference_mode
 def convert_stable_diffusion_to_tflite(
+    ckpt_format: str,
```
Comment (haozha111) [inline ai_edge_torch/generative/examples/stable_diffusion/convert_to_tflite.py:71]: not needed if we only support 1.

File: ai_edge_torch/generative/examples/stable_diffusion/convert_to_tflite.py [hunk 8]
```diff
@@ -24,38 +26,92 @@
...
-  clip_model = clip.CLIP(clip.get_model_config())
-  loader = loading_utils.ModelLoader(clip_ckpt_path, clip.TENSOR_NAMES)
-  loader.load(clip_model, strict=False)
+  strict_loading = ckpt_format == 'pytorch'
 
-  encoder = Encoder()
-  encoder.load_state_dict(torch.load(encoder_ckpt_path))
+  clip_model = clip.CLIP(
+      clip.get_model_config(qkv_fused_interleaved=(ckpt_format == 'pytorch'))
+  )
```
Comment (haozha111) [inline ai_edge_torch/generative/examples/stable_diffusion/convert_to_tflite.py:85]: We can potentially simplify the code here if we just support the standard HF checkpoint. The pytorch checkpoint is from a much unpopular repo, and having that support here may not be very useful in the longer term..

File: ai_edge_torch/generative/examples/stable_diffusion/decoder.py [hunk 9]
```diff
@@ -288,8 +415,10 @@ def get_model_config() -> unet_cfg.AutoEncoderConfig:
           output_proj_use_bias=True,
           enable_kv_cache=False,
           qkv_transpose_before_split=True,
+          qkv_fused_interleaved=False,
           rotary_percentage=0.0,
       ),
+      enable_hlfb=True,
```
Comment (haozha111) [inline ai_edge_torch/generative/examples/stable_diffusion/decoder.py:421]: if not specified, this should be true by default?

File: ai_edge_torch/generative/examples/stable_diffusion/README.md [hunk 10]
```diff
@@ -2,13 +2,33 @@
...
 
 ## Convert PyTorch to TFLite model
-1. Download PyTorch stable diffusion model [stable-diffusion-pytorch](https://github.com/kjsman/stable-diffusion-pytorch)
-1. Unzip the downloaded model weight into `$HOME/Downloads/stable_diffusion_data`
-1. Run `convert_to_tflite.py`. This will convert the PyTorch models into TFLite models. The stable diffusion model has four components: CLIP (text embedding), encoder, diffusion and decoder models. Each component is converted to a single TFLite model file.
+The example provides source checkpoints mapping, from original HuggingFace repo to reauthored PyTorch model.
+
+### SafeTensors model checkpoints from original HuggingFace repo
+1. Clone original HuggingFace stable diffusion repo [runwayml/stable-diffusion-v1-5](https://huggingface.co/runwayml/stable-diffusion-v1-5)
+2. Run `convert_to_tflite.py` and use `v1-5-pruned-emaonly.safetensors` as the source checkpoints for the conversion script. The stable diffusion model has four components: CLIP (text embedding), encoder, diffusion and decoder models. Each component is converted to a single TFLite model file. Notice that the conversion of encoder model is not supported yet.
```
Comment (haozha111) [inline ai_edge_torch/generative/examples/stable_diffusion/README.md:9]: curious why we are using this checkpoint? is this the standard SD 1.5 HF checkpoint?

File: ai_edge_torch/generative/examples/stable_diffusion/decoder.py [hunk 11]
```diff
@@ -22,129 +22,131 @@
 import ai_edge_torch.generative.layers.unet.model_config as unet_cfg
 import ai_edge_torch.generative.utilities.stable_diffusion_loader as stable_diffusion_loader
 
-TENSORS_NAMES = stable_diffusion_loader.AutoEncoderModelLoader.TensorNames(
-    post_quant_conv="0",
-    conv_in="1",
+TENSOR_NAMES = stable_diffusion_loader.AutoEncoderModelLoader.TensorNames(
```
Comment (haozha111) [inline ai_edge_torch/generative/examples/stable_diffusion/decoder.py:25]: just curious, will this mapping change if we are using a different SD 1.5 checkpoint?

Comment (yichunk) [reply ai_edge_torch/generative/examples/stable_diffusion/decoder.py:25]: The tensor names are defined by the model implementation. As long as it use the same model implementation, in this case, `StableDiffusionPipeline` in `diffuser` from HuggingFace, the mapping would be the same.

**Review by haozha111 — DISMISSED** [review]
just to confirm, have you tested by running a few iterations and check the generate image?

**Comment by yichunk** [comment]
> just to confirm, have you tested by running a few iterations and check the generate image?

yes, verify it works with generated image.

**Review by haozha111 — APPROVED** [review]

**Pull request merged**